plugins/daapclient/__init__.py

```diff
@@ -307,12 +307,13 @@ def hmfactory(menu, parent, context):
             
         self.history = DaapHistory(5, menu=hmenu, callback=self.connect_share)
 
-    def connect_share(self, obj, (name, address, port, svc)):
+    def connect_share(self, obj, *args):
```

Comment:
Ok, it seems you are right about this one.

I didn't yet actually tried with python3, because there is still some packaging issue with python3.5 (exaile's Makefile does not installing 3.5's __cache__  dirs with bytecode in them, only py[co] bytecode that is produced with pythons < 3.5), but with these changes it compiles to bytecode without any errors. So I believe it is the way to go. 

I done exaile-devel FreeBSD port (from git master) and it works flawlessly (I'm using it about month already). But I want it to use python3, so here is the changes.

I'll try to overcome __cache__ issue and make it a try with python3 today. Will let you know.